numpy-stubs/__init__.pyi

```diff
@@ -92,13 +92,17 @@ _DtypeLike = Union[
 
 _NdArraySubClass = TypeVar("_NdArraySubClass", bound=ndarray)
 
-_ArrayLike = TypeVar("_ArrayLike")
+class _SupportsArray(Protocol):
+    @overload
+    def __array__(self, __dtype: DtypeLike = ...) -> ndarray: ...
+    @overload
+    def __array__(self, dtype: Optional[DtypeLike] = ...) -> ndarray: ...
```

Comment:
Is the ``Optional`` in ``Optional[DtypeLike]`` not redundant here?
Considering ``None`` is already included the union of ``DtypeLike``.